app/helpers/object_link_helper.rb

```diff
@@ -129,7 +127,7 @@ def user_list(title, users = [])
 
     title = users.count > 1 ? title.to_s.pluralize.to_sym.t : title.t
     links = users.map { |u| user_link(u, u.legal_name) }
-    title + ": " + links.safe_join(", ")
+    "#{title}: #{links.safe_join(", ")}".html_safe
```

Comment:
`html_safe` is needed, else there are test failures like that shown in part below.
The interpolated string is at least as safe as the old concatenated string. 
(My only question is whether a translation could contain unsafe stuff.)
```
 FAIL["test_show_name", #<Minitest::Reporters::Suite:0x000056420fe34548 @name="NameControllerTest">, 61.41838221199578]
 test_show_name#NameControllerTest (61.42s)
        Unsafe HTML found!Here's the appropriate part of the HTML page:
         ...me/288002050?version=1">Previous Version </a><br/>
...

        	<p>First person to use this name on MO: ***HERE***&lt;a href=&quot;/observer/show_user/241228755&quot;&gt;Rolf Singer&lt;/a&gt;&lt;br/&gt;</p>
...
        test/check_for_unsafe_html.rb:23:in `check_for_unsafe_html!'
        test/functional_test_case.rb:27:in `get'
        test/controller_extensions.rb:113:in `get_with_dump'
        test/controllers/name_controller_test.rb:243:in `test_show_name'
``

Reply:
The .t mechanism should guarantee it is safe, and in fact should return an html_safe string.  So this has to have something to do with how we're concatenating the string.  I don't remember the subtleties of it any more, I'll have to look up how to concatenate strings on-line...

Reply:
Hmmm, what about

    [title, ": ", links.safe_join(", ")].safe_join

Does that work?  Really too bad string interpolation doesn't "know about" html_safe.  But honestly, if

    title + ": " + links.safe_join(", ")

works, then that is *much* easier on the eyes.  I'd just as soon just ignore rubocop on this one.

Reply:
Yeah, it looks like you *have* to concatenate carefully: safe_buffer on left, then either unsafe or safe strings on right.  It will escape unsafe strings as it concatenates them.  But the string interpolation starts with an unsafe string on the left, essentially:

    "" + title + ": " + links.safe_join(", ")

That doesn't work, it has to be:

  "".html_safe + title + ": " + links.safe_join(", ")

No way to do that with string interpolation.  So we cannot ever follow rubocop in these cases.  Or we have to cast the whole thing html_safe, which is a huge no-no.

At least that's how I understand it.  I have not tested these assertions to prove that I understand it all correctly.

Reply:
> Yeah, it looks like you _have_ to concatenate carefully: safe_buffer on left, then either unsafe or safe strings on right. It will escape unsafe strings as it concatenates them. But the string interpolation starts with an unsafe string on the left, essentially:
> 
> ```
> "" + title + ": " + links.safe_join(", ")
> ```
> 
> That doesn't work, it has to be:
> 
> "".html_safe + title + ": " + links.safe_join(", ")
> 
> No way to do that with string interpolation. So we cannot ever follow rubocop in these cases. Or we have to cast the whole thing html_safe, which is a huge no-no.
> 
> At least that's how I understand it. I have not tested these assertions to prove that I understand it all correctly.

Thanks for looking into this.
I'll follow your suggestion of using concatenation and turning off the cop for that line.

Reply:
Another thought: 
Would sanitizing the interpolated string break things?
```ruby
sanitize(
  "#{title}: #{links.safe_join(", ")}"
)
```

Reply:
On second thought, I'll use `sanitize` **unless you think it will break something**
It seems so unnecessary, but I think it will be easier on the eye then the combination of:
- turning the cop off
- explaining why the cop is off
- turning it back on

Reply:
Arg, see, this is why I think of rubocop as being more than just a suggestion! It coerces us into using less efficient, harder-to-read and harder-to-maintain constructions.  The original construction was optimal.  We not only end up in a worse place, but we waste all this time trying to puzzle out how to get around this unnecessary roadblock!

Ahem, rant done.  Okay, how about this compromise?

  safe_concat(title, ": ", links.safe_join(", "))

  # General helper:
  def safe_concat(*args)
    args.safe_join("")
  end

This saves us from running sanitize.  (Our views already are the weak point in our performance, I'd rather not pile on even more needlessly.  This sort of construction is probably used all over the place.)

And I think it is cleaner than the ugly string interpolation.  Compare:

  title + ": " + links.safe_join(", ")
  safe_concat(title, ": ", links.safe_join(", "))
  sanitize("#{title}: #{links.safe_join(", ")}")

You are welcome to disagree, of course.  I'm just trying to avoid the unnecessary sanitize even though, I admit, this one call is not going to make the tiniest difference in any real benchmark.  But multiply it by 10, 100, 1000.  When does it become a problem?  I get stuck on: "but, but, but... it's just not necessary! we have just as good ways to do it that don't cause a performance hit!"

Oh hey, there is yet another option no one seems even to have considered!  Can we simply turn the cop off??  Is there really a compelling reason to force these changes to string interpolation?  Can we not simply make a policy decision that either construction is acceptable?